lib/ecto/adapters/mysql/connection.ex

```diff
@@ -478,6 +463,10 @@ if Code.ensure_loaded?(Mariaex.Connection) do
                 if_do(index.concurrently, "LOCK=NONE")])
     end
 
+    def execute_ddl({:create_if_not_exists, %Index{}}) do
+      raise Ecto.QueryError, query: "CREATE INDEX IF NOT EXISTS", message: "Not supported in MySQL"
```

Comment:
No, it should be something like this: https://github.com/gjaldon/ecto/blob/rollback-reference-fix/lib/ecto/adapters/mysql/connection.ex#L488 :)